cli/src/maintenance.rs

```diff
@@ -576,32 +624,49 @@ impl SolidoState {
     /// or if some joker donates to one of the stake accounts we can use the same function
     /// to claim these rewards back to the reserve account so they can be re-staked.
     pub fn try_withdraw_inactive_stake(&self) -> Option<(Instruction, MaintenanceOutput)> {
-        for (validator, stake_accounts) in self
-            .solido
-            .validators
-            .entries
-            .iter()
-            .zip(self.validator_stake_accounts.iter())
-        {
-            let current_balance = stake_accounts
+        for (validator, stake_accounts, unstake_accounts) in izip!(
+            self.solido.validators.entries.iter(),
+            self.validator_stake_accounts.iter(),
+            self.validator_unstake_accounts.iter()
+        ) {
+            let current_stake_balance = stake_accounts
                 .iter()
                 .map(|(_addr, detail)| detail.balance.total())
                 .sum::<token::Result<Lamports>>()
                 .expect("If this overflows, there would be more than u64::MAX staked.");
 
-            if current_balance > validator.entry.stake_accounts_balance {
-                let expected_difference = (current_balance
-                    - validator.entry.stake_accounts_balance)
-                    .expect("Does not overflow because current > entry.balance.");
-                // If the expected difference is less than some defined amount
-                // of Lamports, we don't bother withdrawing. We try to do this
-                // so we don't pay more for fees than the amount that we'll
-                // withdraw.
-                if expected_difference < SolidoState::MINIMUM_WITHDRAW_AMOUNT {
-                    continue;
+            let expected_difference_stake =
+                if current_stake_balance > validator.entry.stake_accounts_balance {
```

Comment:
As `current_stake_balance` only sums over stake accounts, not unstake accounts, this should compare to `validator.entry.effective_stake_balance()`, not to `stake_accounts_balance`.

(I’m more and more wondering if we should’t have stored only the sum of the stake accounts in `stake_accounts_balance` because this appears to be a bit subtle ... but let’s not change that so close before the launch.)

Reply:
Yeah, it's a bit confusing. We can change it later

Reply:
> We can change it later

Not easily; it’s stored in the on-chain state, changing that after v1 will not be worth the trouble, and changing it now before v1 is not worth the risk in my opinion, so we are stuck with it. But it’s not the end of the world; probably if we change it we find other places that become subtle.